crates/polars-plan/src/plans/optimizer/set_order/expr_pushdown.rs

```diff
@@ -216,61 +222,24 @@ pub fn get_frame_observing(
         },
         AExpr::AnonymousFunction { input, options, .. }
         | AExpr::Function { input, options, .. } => {
-            // By definition, does not observe any input so cannot observe any ordering.
-            if input.is_empty() {
-                return Ok(
-                    if options.flags.returns_scalar() || options.flags.is_output_unordered() {
-                        O::None
-                    } else {
-                        O::Independent
-                    },
-                );
-            }
+            let input_ordering = if input.is_empty() {
```

Comment:
This is not correct. An expression that does not have any input does not have unordered output by definition.

See what checks were here before, otherwise we must assume that it is `Independent`